lab_1/main.py

```diff
@@ -116,22 +105,19 @@ def compare_profiles(
     :param top_n: a number of the most common words
     :return: the distance
     """
-    if (
-        isinstance(unknown_profile, dict)
-        and isinstance(profile_to_compare, dict)
-        and isinstance(top_n, int)
-    ):
-        top_1 = get_top_n_words(unknown_profile.get("freq"), top_n)
-        top_2 = get_top_n_words(profile_to_compare.get("freq"), top_n)
-        if top_1 and top_2:
-            common = set(top_1).intersection(set(top_2))
-            score = round(len(common) / len(top_2), 2)
-            return score
-    return None
+    if (not isinstance(unknown_profile, dict)
+            or not isinstance(profile_to_compare, dict)
+            or not isinstance(top_n, int)):
+        return None
+    top_1 = get_top_n_words(unknown_profile["freq"], top_n)
```

Comment:
better to name like `first_top` here and further
it's not a good style to use numbers in names